xontrib/brace_expansion.py

```diff
@@ -21,8 +21,12 @@
 
 @events.on_transform_command
 def expand_braces(cmd, **_):
+    if '[' in cmd or '$(' in cmd or '@(' in cmd:
+        return cmd
+
     parts = cmd.split()
     expanded_cmds = []
+    requires_expansion = False
 
     for part in parts:
         if "{" in part and "}" in part:
```

Comment:
A regular expression might be more reliable and could exclude nested braces. Could be like this:

```python
```python
import re

pattern = r"\{([a-zA-Z],?)+\}"
test_strings = ["{a,o,e,u}", "{x,y,z}", "{aoeu}", "{a,o,e,u,{x},y}"]

for test in test_strings:
    if re.match(pattern, test):
        print(f"Match: {test}")
    else:
        print(f"No match: {test}")
```